packages/library-legacy/src/errors.ts

```diff
@@ -44,25 +44,32 @@ export class SendTransactionError extends Error {
 
     this.signature = signature;
     this.transactionMessage = transactionMessage;
-    this.logs = logs ? logs : undefined;
+    this.transactionLogs = logs ? logs : undefined;
   }
 
   get transactionError(): {message: string; logs?: string[]} {
     return {
       message: this.transactionMessage,
-      logs: Array.isArray(this.logs) ? this.logs : undefined,
+      logs: Array.isArray(this.transactionLogs)
+        ? this.transactionLogs
+        : undefined,
     };
   }
 
+  /* @deprecated Use `await getLogs()` instead */
+  get logs() {
+    return this.transactionLogs as string[] | undefined;
```

Comment:
This may not be true, right? It could be a `Promise`? I think you have to just return `undefined` if `'then' in this.transactionLogs`

Reply:
Thank you. I added that check.

Reply:
Hopefully now it will not be a breaking change anymore.